mllib/src/main/scala/org/apache/spark/mllib/stat/MultivariateOnlineSummarizer.scala

```diff
@@ -233,7 +233,7 @@ class MultivariateOnlineSummarizer extends MultivariateStatisticalSummary with S
   override def numNonzeros: Vector = {
     require(totalWeightSum > 0, s"Nothing has been added to this summarizer.")
```

Comment:
Is it better to check totalCnt here instead?
Also, could you add a simple test for this case?